nixos/modules/virtualisation/amazon-image.nix

```diff
@@ -88,7 +88,7 @@ in
 
     fileSystems."/".device = "/dev/disk/by-label/nixos";
 
-    boot.initrd.kernelModules = [ "xen-blkfront" ];
+    boot.initrd.kernelModules = [ "xen-blkfront" "virtio" "virtio_blk" "virtio_pci" "virtio_ring" ];
```

Comment:
I should probably make these modules conditional too, since an actual EC2 instance doesn't need them (but isn't hurt by them either).